src/H5Eint.c

```diff
@@ -283,20 +283,11 @@ H5E_term_package(void)
     nstk = H5I_nmembers(H5I_ERROR_STACK);
 
     if ((ncls + nmsg + nstk) > 0) {
-        /* Clear the default error stack. Note that
-         * the following H5I_clear_type calls do not
-         * force the clears and will not be able to
-         * clear any error message IDs that are still
-         * in use by the default error stack unless we
-         * clear that stack manually.
-         *
-         * Error message IDs will typically still be
-         * in use by the default error stack when the
-         * application does H5E_BEGIN/END_TRY cleanup
-         * at the very end.
-         */
+        /* Clear the global error stack if it exists.
+         * Thread-local error stack cleanup is handled in H5TS. */
+#ifndef H5_HAVE_THREADSAFE
```

Comment:
I don't think this should be skipped for threadsafe builds.

Reply:
At the time `H5TS_term_package()` is performed during a threadsafe build, H5TS has to clean up the thread info node containing the stack, so it must also clean up the stack. That's done before H5E_term_package(), so its stack is already cleaned, so H5E can't clean it up again. I agree this is a bad pattern. It could be resolved by splitting H5TS termination into two routines.

Reply:
I can't see how the H5TS shutdown could possibly occur before the H5E package shuts down.